test/sandbox/config/routes.rb

```diff
@@ -32,7 +32,7 @@
   get :inherited_sidecar, to: "integration_examples#inherited_sidecar"
   get :inherited_from_uncompilable_component, to: "integration_examples#inherited_from_uncompilable_component"
 
-  constraints(lambda { |request| request.env["warden"].authenticate! }) do
+  constraints(lambda { |request| Thread.current[:constrained_route_enabled] }) do
```

Comment:
I feel hesitant about this change. This does not feel like an equivalent test. Maybe we could use an integration test instead?